src/code_80091750.c

```diff
@@ -486,11 +486,11 @@ char *gDebugCourseNames[] = {
 };
 
 const s8 gPerCupIndexByCourseId[] = {
-    3, 2, 3, 2,
-    1, 1, 2, 2,
-    0, 1, 0, 3,
-    1, 3, 0, 1,
-    3, 2, 0, 0,
+    SPECIAL_CUP, STAR_CUP, SPECIAL_CUP, STAR_CUP,
+    FLOWER_CUP, FLOWER_CUP, STAR_CUP, STAR_CUP,
+    MUSHROOM_CUP, FLOWER_CUP, MUSHROOM_CUP, SPECIAL_CUP,
+    FLOWER_CUP, SPECIAL_CUP, MUSHROOM_CUP, FLOWER_CUP,
+    SPECIAL_CUP, STAR_CUP, MUSHROOM_CUP, MUSHROOM_CUP,
```

Comment:
Yesterday I noticed something doesn't seem right about this array.

Choco mountain is course id 1 which we'd expect as flower cup here. But if 2 = star cup. That seems weird. This might not be cupIndexByCourseId but perhaps cupIndexByCupId or something weird like that.